src/modules/dashboard/dashboard.service.ts

```diff
@@ -0,0 +1,321 @@
+import { Injectable } from '@nestjs/common';
+import { Prisma, UserRole } from '@prisma/client';
+import { subMonths } from 'date-fns';
+
+import { calculatePercentage } from '../../utils/helpers';
+import { MatchesService } from '../matches/matches.service';
+import { NotesService } from '../notes/notes.service';
+import { OrganizationSubscriptionsService } from '../organization-subscriptions/organization-subscriptions.service';
+import { PlayersService } from '../players/players.service';
+import { PrismaService } from '../prisma/prisma.service';
+import { ReportsService } from '../reports/reports.service';
+import { CurrentUserDto } from '../users/dto/current-user.dto';
+import { UsersService } from '../users/users.service';
+import { DashboardDto } from './dto/dashboard.dto';
+import {
+  transformMatchSubscriptions,
+  transformObservationSubscriptions,
+  transformPlayerSubscriptions,
+} from './transform-subscriptions';
+
+@Injectable()
+export class DashboardService {
+  constructor(
+    private readonly prisma: PrismaService,
+    private readonly organizationSubscriptionsService: OrganizationSubscriptionsService,
+    private readonly reportsService: ReportsService,
+    private readonly notesService: NotesService,
+    private readonly matchesService: MatchesService,
+    private readonly playersService: PlayersService,
+    private readonly usersService: UsersService,
+  ) {}
+
+  // PM-ScoutManager | ADMIN | 'SCOUT'
+  private async getCommonData(user: CurrentUserDto) {
+    const getMatchesCount = (filters: Prisma.MatchWhereInput) =>
+      this.matchesService.getCount({ accessFilters: filters });
+    const getReportsCount = (filters: Prisma.ReportWhereInput) =>
+      this.reportsService.getCount({ accessFilters: filters });
+    const getNotesCount = (filters: Prisma.NoteWhereInput) =>
+      this.notesService.getCount(filters);
+
+    const data: DashboardDto = { user };
+
+    const monthAgoDate = subMonths(new Date(), 1);
+
+    const rolesScope: UserRole[] =
+      user.role === 'SCOUT'
+        ? ['SCOUT']
+        : ['PLAYMAKER_SCOUT', 'ADMIN', 'PLAYMAKER_SCOUT_MANAGER'];
+
+    const recentScopedObservationsWhere:
+      | Prisma.NoteWhereInput
+      | Prisma.ReportWhereInput = {
+      createdAt: { gte: monthAgoDate },
+      author: {
+        role: {
+          in: rolesScope,
+        },
+      },
+    };
+
+    const recentUserObservationsWhere:
+      | Prisma.NoteWhereInput
+      | Prisma.ReportWhereInput = {
+      createdAt: { gte: monthAgoDate },
+      authorId: user.id,
+    };
+
+    // reports
+    const totalUserReportsPromise = getReportsCount({
+      authorId: user.id,
+    });
```

Comment:
```suggestion
    const totalUserReportsPromise = this.reportsService.getCount({ accessFilters: { authorId: user.id } })
```

Why don't we do it like this?